src/RoomsSDKAdapter.js

```diff
@@ -134,4 +153,138 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Returns an array of IDs of the most recent activities in a conversation up to the specified limit.
+   *
+   * @param {object} Params.ID ID for the room
+   * @param {object} Params.earliestActivityDate  Get all child activities before this date
+   * @returns {Promise} Resolves with array of activities
+   * @private
+   */
+  async fetchActivities({
+    ID,
+    earliestActivityDate,
+  }) {
+    const {limit} = this;
+    const conversationId = deconstructHydraId(ID).id;
+
+    logger.debug('ROOM', 'fetchActivities()', ['called with', {
+      ID,
+      limit,
+      earliestActivityDate,
+    }]);
+
+    if (!FETCHED_CONVERSATIONS) {
+      await this.datasource.internal.conversation.list();
+      FETCHED_CONVERSATIONS = true;
+    }
+
+    return this.datasource.internal.conversation.listActivities({
+      conversationId,
+      limit: limit + 1,
+      lastActivityFirst: true,
+      maxDate: earliestActivityDate === null ? undefined : earliestActivityDate,
+    });
+  }
+
+  /**
+   * Returns `true` if there are more activities to load from the room of the given ID.
+   * Otherwise, it returns `false`.
+   *
+   * @param {string} ID ID of the room for which to verify activities.
+   * @returns {boolean} `true` if room has more activities to load, `false` otherwise
+   */
+  hasMoreActivities(ID) {
+    const roomObservable = this.roomObservables[ID];
```

Comment:
The convention is to use `$` at the end of the variable name to indicate observable
```suggestion
    const room$ = this.roomObservables[ID];
```